### PR TITLE
💪  Select returns picked types!

### PR DESCRIPTION
**Pull request by nicolasgramlich**
Followup: https://github.com/bluebirds-blue-jay/sequelize-service/pull/5
- [x] Code
- [x] Tests
- [x] Docs

### CODE REVIEW

File: test/classes/sequelize-service.test.ts [hunk 1]
```diff
@@ -210,14 +209,28 @@ describe('SequelizeService', function () {
       expect(user).to.have.keys(['email', 'id']); // ID is auto selected
     });
 
+    it('should only resolve types for what was selected (+ auto selected id)', async () => {
+      await userService.create({ email: 'foo', password: 'bar' });
+      const [ user ] = await userService.find({}, { select: ['email'] });
+      expect(user).to.have.keys(['email', 'id']); // ID is auto selected
+      expect(user).to.not.have.keys(['first_name']);
+      // Note: user.first_name should not even compile!
```
Comment (nicolasgramlich) [inline test/classes/sequelize-service.test.ts]: @SylvainEstevez `user.isAdult` compiles. That's not intended, right?

File: test/classes/sequelize-service.test.ts [hunk 2]
```diff
@@ -210,14 +209,28 @@ describe('SequelizeService', function () {
...
+    });
+
+    it('should by default select everything', async () => {
+      await userService.create({ email: 'foo', password: 'bar' });
+      const [ user ] = await userService.find({});
+      expect(user).to.have.keys(['id', 'email', 'first_name', 'last_name', 'date_of_birth', 'lucky_number', 'password', 'password_last_updated_at', 'updated_at', 'created_at']);
```
Comment (nicolasgramlich) [inline test/classes/sequelize-service.test.ts]: @SylvainEstevez (Same here) `user.isAdult` compiles. That's not intended, right?

File: test/classes/sequelize-service.test.ts (hunk 1 above)
Comment (SylvainEstevez) [reply test/classes/sequelize-service.test.ts]: Definitely not, since it’s not in compute

Comment (nicolasgramlich) [reply test/classes/sequelize-service.test.ts]: (Note: it doesn't have that actual key, typescript just thinks so)

Yeah I think there is a "lucky default" going on somewhere. Will fix...

Comment (nicolasgramlich) [reply test/classes/sequelize-service.test.ts]: `compute` in followup PR: https://github.com/bluebirds-blue-jay/sequelize-service/pull/5

File: src/classes/sequelize-service.ts [hunk 3]
```diff
@@ -96,29 +96,29 @@ export class SequelizeService<W, R extends W, C extends {} = {}> extends Service
...
     return object || null;
   }
 
-  public async findByPrimaryKey(pk: string | number, options: TFindByPrimaryKeyOptions<R, C> = {}): Promise<R & C> {
+  public async findByPrimaryKey<S extends keyof R>(pk: string | number, options: TFindByPrimaryKeyOptions<R, C, S> = {}): Promise<Pick<R, S> & C> {
     return await this.findOne({ [this.getPrimaryKeyField()]: pk } as any, options);
   }
 
-  public async findByPrimaryKeys(pks: string[] | number[], options: TFindByPrimaryKeyOptions<R, C> = {}): Promise<Collection<R & C>> {
+  public async findByPrimaryKeys<S extends keyof R>(pks: string[] | number[], options: TFindByPrimaryKeyOptions<R, C, S> = {}): Promise<Collection<Pick<R, S> & C>> {
```
Comment (SylvainEstevez) [inline src/classes/sequelize-service.ts]: No `= R` as a default ?

Comment (nicolasgramlich) [reply src/classes/sequelize-service.ts]: Somehow that's the default. I had to explicitly set it to undefined to remove it: https://github.com/bluebirds-blue-jay/sequelize-service/pull/5/files#diff-4d1b16e85980be420ff08d71cd67d774R55

**Pull request merged**